vividus-plugin-rest-api/src/main/java/org/vividus/http/HttpRequestExecutor.java

```diff
@@ -0,0 +1,135 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import javax.inject.Named;
+
+import org.apache.http.ConnectionClosedException;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.client.IHttpClient;
+import org.vividus.http.exception.HttpRequestBuildException;
+import org.vividus.softassert.ISoftAssert;
+
+/**
+ * Executor of HTTP requests which supports some exceptions handling, test context releasing
+ * and requests repeat with quit conditions.
+ */
+@Named
+public class HttpRequestExecutor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpRequestExecutor.class);
+
+    private final IHttpClient httpClient;
+    private final HttpTestContext httpTestContext;
+    private final ISoftAssert softAssert;
+
+    public HttpRequestExecutor(IHttpClient httpClient, HttpTestContext httpTestContext, ISoftAssert softAssert)
+    {
+        this.httpClient = httpClient;
+        this.httpTestContext = httpTestContext;
+        this.softAssert = softAssert;
+    }
+
+    /**
+     * Executes HTTP request and frees context request entity and headers.
+     * Request is executed safely for unchecked exceptions: HttpRequestBuildException, ConnectionClosedException.
+     * Assertion fails if exceptions occurred.
+     * @param httpMethod HttpMethod to execute
+     * @param endpoint Request endpoint
+     * @param relativeURL Relative URL - optional
+     * @throws IOException If an input or output exception occurred
+     */
+    public void executeHttpRequest(HttpMethod httpMethod, String endpoint, Optional<String> relativeURL)
+            throws IOException
+    {
+        executeHttpRequest(httpMethod, endpoint, relativeURL, response -> false);
+    }
+
+    /**
+     * Executes HTTP request <i>while predicate condition is true</i>
+     * and frees context request entity and headers afterwards.
+     * Request is executed safely for unchecked exceptions: HttpRequestBuildException, ConnectionClosedException.
+     * Assertion fails if exceptions occurred.
+     * @param httpMethod HttpMethod to execute
+     * @param endpoint Request endpoint
+     * @param relativeURL Relative URL - optional
+     * @param repeatRequest Predicate condition to check response
+     * @throws IOException If an input or output exception occurred
+     */
+    public void executeHttpRequest(HttpMethod httpMethod, String endpoint, Optional<String> relativeURL,
```

Comment:
No reason to use Optional ? 
I found only `relativeUrl.orElse(null)` ?

Reply:
https://github.com/vividus-framework/vividus/pull/59#discussion_r337352845